In [1]:
import pandas as pd


In [12]:
! pip install statsmodels

You should consider upgrading via the 'C:\Users\Дмитрий\AppData\Local\Programs\Python\Python39\python.exe -m pip install --upgrade pip' command.


In [3]:
! pip install scipy

You should consider upgrading via the 'C:\Users\Дмитрий\AppData\Local\Programs\Python\Python39\python.exe -m pip install --upgrade pip' command.


# Домашнее задание
## Используя информацию по мидиям Петербурга и Магадана, которые представлены в виде двух массивов petersburg и magadan в разделе ANOVA тест, проверьте данные на нормальность и на наличие корреляции. ## BДля этого объедините 2 массива в DataFrame. Обоснуйте выбор теста на корреляцию. Сделайте вывод по гипотезе.
 
 
## Также ответьте на вопрос:
## Какова разница в среднем размере мидии в зависимости от города-производителя. Обоснуйте выбор теста. Сделайте вывод по гипотезе.

In [9]:
import pandas as pd
	## Данные объединены в DataFrame 
petersburg = [0.0974, 0.1352, 0.0817, 0.1016, 0.0968, 0.1064, 0.105]
magadan = [0.1033, 0.0915, 0.0781, 0.0685, 0.0677, 0.0697, 0.0764,
           0.0689]
moul_data = pd.DataFrame([petersburg,magadan] ,index =  ["petersburg","magadan"])
moul_data=moul_data.transpose()
moul_data
# Заполняем пропущенное значение медианой
moul_data.fillna({'petersburg': moul_data['petersburg'].median()}, inplace=True)

display(moul_data)



petersburg  magadan
0      0.0974   0.1033
1      0.1352   0.0915
2      0.0817   0.0781
3      0.1016   0.0685
4      0.0968   0.0677
5      0.1064   0.0697
6      0.1050   0.0764
7      0.1016   0.0689

##  Какова разница в среднем размере мидии в зависимости от города-производителя

In [23]:
from scipy.stats import ttest_ind
H0 = 'Нет значимой разницы между диаметрами пиццы в разных городах.'
Ha = 'Есть значимая разница между диаметрами пиццы в разных городах.'
def t_test(df):
    print('\n' + "*** Результаты независимого T-теста ***")
    test_results = ttest_ind(moul_data['petersburg'], moul_data['magadan'], equal_var=True)

    p = round(test_results[1],2)

    if p>alpha:
        print(f"{p} > {alpha}. Мы не можем отвергнуть нулевую гипотезу. {H0}")
    else:
        print(f"{p} <= {alpha}. Мы отвергаем нулевую гипотезу. {Ha}")

t_test(moul_data)



*** Результаты независимого T-теста ***
0.0 <= 0.05. Мы отвергаем нулевую гипотезу. Есть значимая разница между диаметрами пиццы в разных городах.


# Проверка данных на нормальность


## Тест Шапиро-Уилка

In [10]:
from scipy.stats import shapiro
_, p = shapiro(moul_data)
print(f"Критерий Шапиро-Уилка: p-value = {p: .3f}")

Критерий Шапиро-Уилка: p-value =  0.101


### Критерий Шапиро-Уилка: p-value =  0.101
### Вывод: p-value больше принятого уровня значимости 0.05, следовательно нулевая гипотеза не отвергается, а распределение размеров раковин можно считать нормальным

# Проверка на наличие корреляции

## Поскольку данные распределены по нормальному закону, следует применять параметрические тесты, в случае проверки на корреляцию непрерывных числовых признаков- это коэффициент корреляции r Пирсона

In [11]:
moul_data
print(moul_data.corr(method="pearson"))

            petersburg   magadan
petersburg    1.000000  0.256665
magadan       0.256665  1.000000



## Вывод :  коэффициент карелации 0.26  показывает об отсутствии связи либо слабой связи

# Сравнение  средных размеров по выборкам 
### Принимаем стандартное значение уровня значимости alpha = 0.05

### Поскольку признаки распределены нормально (необходим параметрический тест), сравниваются только две выборки по одному параметру (не нужен дисперсионный анализ), объем выборок не превышает 30 #######(исключаем z-тест), и выборки независимы (происходят из разных совокупностей), применяем t-test (критерий Стьюдента) для независимых выборок. Предполагаем равенство дисперсий в двух выборках.

In [16]:
from scipy.stats import ttest_ind
print(f'Независимый t-тест: p-value = {p: .3f}')

Независимый t-тест: p-value =  0.003


In [19]:
from statsmodels.stats import weightstats
_ ,p = weightstats.ztest(moul_data['petersburg'], x2=moul_data['magadan'], value=0,alternative='two-sided')
print(float(p))


if p < 0.05:
    print("Мы отвергаем нулевую гипотезу об отсутсвии различий между данными.")
else:
    print("Мы не можем отвергнуть нулевую гипотезу об отсутсвии различий между данными.")

0.0003250100933559849
Мы отвергаем нулевую гипотезу об отсутсвии различий между данными.


### Выводы :  p-value меньше установленного уровня значимости 0.05 (более того, меньше уровня 1%), следовательно есть статистические значимые различия в рамерах раковин. Рассчитаем средние размеры:

In [20]:
a_peter = moul_data['petersburg'].mean()
a_magadan = moul_data['magadan'].mean()
print(f'Средний размер раковины (Петербург): {a_peter}')
print(f'Средний размер раковины (Магадан): {a_magadan}')

Средний размер раковины (Петербург): 0.10321249999999998
Средний размер раковины (Магадан): 0.0780125


### Вывод: мидии из Петербурга в среднем крупнее, различия статистически значимы